### ISSUE TITLE
Aurora eating ram on start up and displaying no interface or lighting effects

### ISSUE THREAD

**Issue by JmLykins** [antonpup/Aurora#749]
### What are you experiencing an issue with?

- [x] Aurora
- [ ] Aurora supported game, (Fill in the game)
- [ ] Aurora Wrapper
- [ ] Aurora Profile
- [ ] Aurora supported device, (Fill in problematic devices)
- [ ] Aurora Scripts
- [ ] Aurora Updater
- [x] Aurora Lighting issue
- [ ] Other, (Fill in another issue here)

### What kind of an issue is it?

- [ ] Crash
- [x] Performance problems
- [ ] Visual bug
- [ ] Localization issue
- [ ] Enhancement suggestion
- [ ] Device issue
- [ ] Other, (Fill in another type of issue here)

### Expected behavior:
Aurora should start up with Windows and load lighting profile without eating ram.

### Actual behavior:
Aurora loads without any interface and starts to hog ram, lighting stays black.

### How often does this problem occur?

This problem occurs... (Every start up. Ending task and restarting aurora fixes problem)

### Please describe the steps to reproduce this issue:
The issue can be reproduced by doing following steps:

1. At start up
2.
3. 

### What is the version of Aurora you are experiencing this issue with?

My version is... (0.6.1-dev2)

### Does the previous version of Aurora have this issue?
The previous version, (0.6.0b), did not have this issue.

### Please include any logs, screenshots, or files that are related to this issue:

For logs, you can use: http://pastebin.com/
For screenshots, you can use: http://imgur.com/
For files, you can use: http://www.tinyupload.com/ , https://www.dropbox.com/ , or attach them to the issue.

**Comment by dfg260** [comment]
Got the exact same problem too.
Logs: https://pastebin.com/zgCX5PPk

**Comment by NeoDaemon** [comment]
Same problem too.
Even loading the program manually sometimes the process start hogging ram and fails to load the ui.
Logs: https://pastebin.com/4sfaf5Hz

**Comment by lucasvanhalst** [comment]
For me this only happens when aurora starts with windows, if I start it manually it works fine

**Comment by FHeilmann** [comment]
Could it be connected to the fact that aurora requires administrator privileges during startup?

I have the same problem btw.

**Comment by simon-wh** [comment]
@JmLykins @dfg260 @NeoDaemon @lucasvanhalst @FHeilmann  Has anyone experienced this while using v0.6.1-dev3?

**Comment by JmLykins** [comment]
@simon-wh Aurora now starts and loads my keyboard lighting configuration but still no ui and starts to eat ram. Restarting Aurora fixes the problem.

**Comment by lucasvanhalst** [comment]
@simon-wh yep, same issue. Even got windows to say 'Close some programs to free up memory' because it ate all of my available ram and swap space.
~![https://i.imgur.com/dBcN7FK.png]()

My light effects do work now, but the UI icon isn't visible in the taskbar and I can't open it unless I kill it in task manager and restart manually.
Using a G910 + razer deathadder elite.

**Comment by FHeilmann** [comment]
My last sync also happened post 0.6.1-dev3 (I self-compile). I saw that some changes were made to the autostart-with-windows code so I'll sync and retest today.

**Comment by simon-wh** [comment]
I haven't been able to recreate this issue again recently but I have experienced it before using dev builds. If anyone who is self-compiling like @FHeilmann could try and figure out why it is happening when it happens again I'd appreciate it.

**Comment by FHeilmann** [comment]
Freshly synced just now.

unfortunately issue still persists.

Here is my log:

https://pastebin.com/xkwSAdtE

**Comment by FHeilmann** [comment]
Addendum: Just booted my PC again and everything started up as it should've. So it seems the issue only occurs sometimes.

**Comment by dfg260** [comment]
> Has anyone experienced this while using v0.6.1-dev3?

I updated it and sadly nothing changed.
Aurora still starts eating ram even if its manually started.

![](https://i.imgur.com/IVs7KY2.png)
Logs: https://pastebin.com/9UntBm7g

**Comment by simon-wh** [comment]
Could everyone with this issue try [this build](https://www.dropbox.com/s/h534it3b96xfd4t/v0.6.1-dev3%20startup%20test.zip?dl=0)? I've had this running on a few bootups and haven't had the issue again yet. It could still be present as I wasn't experiencing the issue consistently.

**Comment by Iruberiam** [comment]
I've just tried that build and sadly the problem remains. No UI and no tray icon

**Comment by simon-wh** [comment]
@lruberiam Could you post your log file?

**Comment by lucasvanhalst** [comment]
I can confirm that it still has the leak in that build. Where can I find the logs? in the aurora folder there's a folder named 'logs' but it's empty

**Comment by brainbug89** [comment]
> Where can I find the logs? in the aurora folder there's a folder named 'logs' but it's empty

`%appdata%\Aurora\Logs`

**Comment by Iruberiam** [comment]
@simon-wh log posted at https://pastebin.com/jU5SPxXm

**Comment by lucasvanhalst** [comment]
So, have you been able to take a look yet? 

It's kinda bad still ![https://i.imgur.com/vY2LeYE.png](https://i.imgur.com/vY2LeYE.png)

**Comment by simon-wh** [comment]
@lucasvanhalst I've taken plenty of looks but I still haven't been able to figure out the cause. It isn't helped by the fact that I can't consistently replicate the issue.

**Comment by Iruberiam** [comment]
For me, Dev3 has been stable, excluding the initial issues with the missing interface.

**Comment by dfg260** [comment]
> Could everyone with this issue try this build? I've had this running on a few bootups and haven't had the issue again yet. It could still be present as I wasn't experiencing the issue consistently.
It still eats lots of ram but, for me, it stops at ~8,5 - 9,5 GB and then it starts to decrease slowly . It also runs normally so u dont notice it doing that if you have enough ram.
so i guess its partialy fixed!
Logs: https://pastebin.com/Ssrw2VUh

**Comment by dfg260** [comment]
@lucasvanhalst you were right. it filled both ram & swap. but you still don't notice it until your pc bluescreens with a system service exception.
![screenshot](https://i.imgur.com/vmQT91b.png)

**Comment by lucasvanhalst** [comment]
So no luck in reproducing this yet? I've build a new pc in the meantime and it also happens on that one. Perhaps it has to do something with the layers I'm using? [https://i.imgur.com/t0LjKAI.png](https://i.imgur.com/t0LjKAI.png)

**Comment by simon-wh** [comment]
@lucasvanhalst I haven't been able to, no, I'm on a new windows install so I'm going to try and see if I can reproduce it.

**Comment by DogeAmazed** [comment]
I think it's somehow related to the Equalizer Layer (so NAudio), but need to look more into it.
Just had this problem when I had music enabled and started Aurora.

**Comment by lucasvanhalst** [comment]
I'm guessing it's related to the audio indeed. It still happened without the eq layer, but when I tried to restart aurora, it wouldn't properly start if I didn't turn off the audio playing in the background first.

**Comment by Wuestengecko** [comment]
I've set up Visual Studio and done some digging. Fortunately (somewhat), the issue is reproducible with a fresh setup and a clean build of the `v0.6.1-dev3` tag on my machine. I've upgraded to the v141 toolchain instead of the v140 one which was originally mentioned in the project files, though that seems to only affect the C++ wrappers and not the C# code. I also tried checking out the current HEAD of the NAudio subrepo (`v1.8.4-5-g1a99d54`), which did not change a thing.

The funny thing about this is, that it seems to not be audio-related at all:

![desktop screenshot 2018 01 27 - 02 46 40 35](https://user-images.githubusercontent.com/1579756/35467295-f5b29fa0-030c-11e8-85c4-497f91883ebe.png)

The green background marks the line of code where it'll return to after completing the "external code" (which I can't find any more info about - external SDK dll it seems). When pressing the resume button, waiting some time and breaking again, the IDE will return to this very state again, so it seems that this external code is actually the reason.

After having done all this already, I realized that there have been 50 more commits to the `dev` branch, and apparently one of those has fixed this issue. It started successfully 10 out of 10 times consecutively.
I've also gone ahead and zipped those built files up, you can download them here: [release_v0_6_1-dev3-50-g2b358db.zip](https://github.com/antonpup/Aurora/files/1669712/release_v0_6_1-dev3-50-g2b358db.zip)

I won't blame you for uploading this to Virustotal, in fact I'd congratulate you for being sensible and cross-checking stuff from that person you just met on the Interwebs. ;)

**Comment by simon-wh** [comment]
For everyone experiencing this issue could you try [v0.6.1-dev4](https://github.com/antonpup/Aurora/releases) which seems to fix the issue, according to @Wuestengecko

**Comment by lucasvanhalst** [comment]
@Wuestengecko @simon-wh Thanks for the efffort, it has worked without issues so far.

**Comment by lucasvanhalst** [comment]
Guess this can be closed, haven't had this issue since then
